### PR TITLE
chore: scaffold spfx project using packages intalled globally 

### PR DESCRIPTION
**Pull request by yuqizhou77**
- use globally installed package to do scaffolding if user select this option.
- if required packages could not be found, show error
image

### CODE REVIEW

File: packages/fx-core/resource/package.nls.json [hunk 1]
```diff
@@ -217,6 +217,8 @@
   "plugins.spfx.error.invalidDependency": "Failed to validate package %s",
   "plugins.spfx.error.installDependency": "Failed to install package %s. Learn how to remediate in the [output](command:fx-extension.showOutputChannel).",
   "plugins.spfx.error.noConfiguration": "There's no .yo-rc.json file in your SPFx project, add the configuration file and try again.",
+  "plugins.spfx.error.devEnvironmentNotSetup": " SharePoint Framework development environment has not been set up correctly. Visit [Set up your SharePoint Framework development environment | Microsoft Learn](%s) to learn more.",
+  "_plugins.spfx.error.devEnvironmentNotSetup.comment": "'Microsoft Learn' and 'SharePoint' are the name of products, which should not be localized. ",
```
Comment (sffamily) [inline packages/fx-core/resource/package.nls.json:221]: the name of products -> product brand names

File: packages/fx-core/resource/package.nls.json [hunk 2]
```diff
@@ -189,11 +189,11 @@
   "plugins.spfx.questions.webpartName.error.notMatch": "%s doesn't match pattern: %s",
   "plugins.spfx.questions.packageSelect.title": "SharePoint Framework",
   "plugins.spfx.questions.packageSelect.placeholder": "Select an option for scaffolding",
-  "plugins.spfx.questions.packageSelect.useGlobalPackage.withVersion.label": "Use globally installed SharePoint Framework (%s)",
-  "plugins.spfx.questions.packageSelect.useGlobalPackage.noVersion.label": "Use globally installed SharePoint Framework",
-  "plugins.spfx.questions.packageSelect.useGlobalPackage.detail": "Recommend SharePoint Framework %s or later",
-  "plugins.spfx.questions.packageSelect.installLocally.withVersion.label": "Install the latest SharePoint Framework (%s) locally in Teams Toolkit directory ",
-  "plugins.spfx.questions.packageSelect.installLocally.noVersion.label": "Install the latest SharePoint Framework locally in Teams Toolkit directory ",
+  "plugins.spfx.questions.packageSelect.useGlobalPackage.withVersion.label": "Use globally installed SPFx (%s)",
```
Comment (yuqizhou77) [inline packages/fx-core/resource/package.nls.json:192]: confirmed with SPFx folks that it is OK to use "SPFx"

File: packages/fx-core/resource/package.nls.json [hunk 3]
```diff
@@ -217,6 +217,8 @@
   "plugins.spfx.error.invalidDependency": "Failed to validate package %s",
   "plugins.spfx.error.installDependency": "Failed to install package %s. Learn how to remediate in the [output](command:fx-extension.showOutputChannel).",
   "plugins.spfx.error.noConfiguration": "There's no .yo-rc.json file in your SPFx project, add the configuration file and try again.",
+  "plugins.spfx.error.devEnvironmentNotSetup": " SPFx development environment has not been set up correctly. Visit [Set up your SharePoint Framework development environment | Microsoft Learn](%s) to learn more.",
+  "_plugins.spfx.error.devEnvironmentNotSetup.comment": "'Microsoft Learn' and 'SharePoint' are product branch names, which should not be localized. ",
```
Comment (sffamily) [inline packages/fx-core/resource/package.nls.json:221]: product brand names

File: packages/fx-core/src/component/resource/spfx/utils/constants.ts [hunk 4]
```diff
@@ -25,10 +25,13 @@ export class Constants {
   public static readonly SUPPORTED_NODE_VERSION = ["16"];
   public static readonly SUPPORTED_NPM_VERSION = ["7", "8"];
   public static readonly SPFX_HELP_LINK = "https://aka.ms/teamsfx-spfx-help";
```
[the remark this answers is not in the document]
Comment (yuqizhou77) [reply packages/fx-core/src/component/resource/spfx/utils/constants.ts:27]: yeah, I think it will.

**Review by sffamily — APPROVED** [review]

**Review by HuihuiWu-Microsoft — APPROVED** [review]

**Review by jayzhang — DISMISSED** [review]

**Review by HuihuiWu-Microsoft — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

**Pull request merged**